src/stories/ActionList2.stories.tsx

```diff
@@ -0,0 +1,1278 @@
+            </ActionList.Item>
+          </ActionList.Group>
+          <ActionList.Divider />
+          <ActionList.Group>
+            <ActionList.Item style={{fontWeight: 'bold'}}>
+              <ActionList.LeadingVisual>
+                <FilterIcon />
+              </ActionList.LeadingVisual>
+              Save sort and filters to current view
+            </ActionList.Item>
+            <ActionList.Item style={{fontWeight: 'bold'}}>
+              <ActionList.LeadingVisual>
+                <FilterIcon />
+              </ActionList.LeadingVisual>
+              Save sort and filters to new view
+            </ActionList.Item>
+          </ActionList.Group>
+          <ActionList.Divider />
+          <ActionList.Group sx={{backgroundColor: 'accent.emphasis'}}>
+            <ActionList.Item sx={{color: 'white'}}>
+              <ActionList.LeadingVisual sx={{color: 'white'}}>
+                <GearIcon />
+              </ActionList.LeadingVisual>
+              View settings
+            </ActionList.Item>
+          </ActionList.Group>
+        </ActionList>
+      </ErsatzOverlay>
+    </>
+  )
+}
+ComplexListFullVariantStory.storyName = 'Complex List — Full Variant'
+
+type ReactRouterLikeLinkProps = {to: string; children: React.ReactNode}
+const ReactRouterLikeLink = forwardRef<HTMLAnchorElement, ReactRouterLikeLinkProps>(
+  ({to, ...props}: {to: string; children: React.ReactNode}, ref) => {
+    // eslint-disable-next-line jsx-a11y/anchor-has-content
+    return <a ref={ref} href={to} {...props} />
+  }
+)
+
+const NextJSLikeLink = forwardRef(
+  ({href, children}: {href: string; children: React.ReactNode}, ref): React.ReactElement => {
+    const child = React.Children.only(children)
+    const childProps = {
+      ref,
+      href
+    }
+    return <>{React.isValidElement(child) ? React.cloneElement(child, childProps) : null}</>
+  }
+)
+
+export function LinkItemStory(): JSX.Element {
+  return (
+    <>
+      <h1>List with LinkItem</h1>
+      <ErsatzOverlay>
+        <ActionList showDividers>
+          <ActionList.Item>
+            <ActionList.LeadingVisual>
+              <XIcon />
+            </ActionList.LeadingVisual>
+            not a link, just an Item for comparison
+          </ActionList.Item>
+          <ActionList.LinkItem href="https://github.com/primer">
+            <ActionList.LeadingVisual>
+              <LinkIcon />
+            </ActionList.LeadingVisual>
+            ActionList.LinkItem
+          </ActionList.LinkItem>
+          <ActionList.LinkItem href="https://github.com/primer" target="_blank" rel="noopener noreferrer">
+            <ActionList.LeadingVisual>
+              <LinkIcon />
+            </ActionList.LeadingVisual>
+            ActionList.LinkItem with anchor attributes
+          </ActionList.LinkItem>
+          <ActionList.LinkItem
+            as={ReactRouterLikeLink}
+            to="?path=/story/composite-components-actionlist2--simple-list-story"
```

Comment:
help wanted: How do we add type support for arbitrary components and their props?